functests/ptp/ptp.go

```diff
@@ -108,4 +97,472 @@ var _ = Describe("ptp", func() {
+
+			status := client.Client.ConfigV1Interface.RESTClient().Delete().AbsPath(ptpConfigSlave.GetSelfLink()).Do()
+			Expect(status.Error()).NotTo(HaveOccurred(), fmt.Sprint("Can not delete PTP config"))
+
+			ptpConfigTest := mutateProfile(ptpConfigSlave, ptpConfigName, nodes.Items[0].Name)
+
+			status = client.Client.ConfigV1Interface.RESTClient().Post().AbsPath(path).
+				Resource("ptpconfigs").Body(ptpConfigTest).Context(context.TODO()).Do()
+			Expect(status.Error()).NotTo(HaveOccurred(), fmt.Sprint("PTP config creation Error"))
+
+			testPtpPod, err = getPtpPodOnNode(nodes.Items[0].Name)
+			Expect(err).NotTo(HaveOccurred())
+
+			testPtpPod, err = replaceTestPod(testPtpPod, 60)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		var _ = Context("Check if Node has Profile", func() {
+			//25743
+			It("PTP daemon can apply match rule based on nodeLabel", func() {
+				err := waitUntilLogIsDetected(testPtpPod, 180, "Profile Name: test")
+				Expect(err).NotTo(HaveOccurred(), "Error to find needed log due to %s", err)
+			})
+		})
+		AfterEach(func() {
+			status := client.Client.ConfigV1Interface.RESTClient().Delete().AbsPath(path + "/test").Do()
+			Expect(status.Error()).NotTo(HaveOccurred(), fmt.Sprint("Can not delete PTP config"))
+
+			ptpConfigSlave.ObjectMeta.SetResourceVersion("")
+			status = client.Client.ConfigV1Interface.RESTClient().Post().AbsPath(path).
+				Resource("ptpconfigs").Body(ptpConfigSlave.DeepCopy()).Context(context.TODO()).Do()
+			Expect(status.Error()).NotTo(HaveOccurred(), fmt.Sprint("PTP slave config recovery Error"))
+			err := waitUntilLogIsDetected(testPtpPod, 180, "Profile Name: slave")
+			Expect(err).NotTo(HaveOccurred(), "Error to find needed log due to %s", err)
+		})
+	})
 })
+
+func podRole(runningPod v1core.Pod, role string) bool {
+	nodeList, err := client.Client.Nodes().List(metav1.ListOptions{
+		LabelSelector: role,
+	})
+	Expect(err).NotTo(HaveOccurred())
+	for NodeNumber := range nodeList.Items {
+		if runningPod.Spec.NodeName == nodeList.Items[NodeNumber].Name {
+			return true
+		}
+	}
+	return false
+}
+
+func execCommand(pod v1core.Pod, command []string) (string, bytes.Buffer) {
+	req := client.Client.CoreV1Interface.RESTClient().
+		Post().
+		Namespace(pod.Namespace).
+		Resource("pods").
+		Name(pod.Name).
+		SubResource("exec").
+		VersionedParams(&v1core.PodExecOptions{
+			Container: pod.Spec.Containers[0].Name,
+			Command:   command,
+			Stdin:     true,
+			Stdout:    true,
+			Stderr:    true,
+			TTY:       true,
+		}, scheme.ParameterCodec)
+	retry := 3
+	for retry > 0 {
+		exec, err := remotecommand.NewSPDYExecutor(client.Client.Config, "POST", req.URL())
+		Expect(err).ToNot(HaveOccurred())
+		var buf bytes.Buffer
+		err = exec.Stream(remotecommand.StreamOptions{
+			Stdin:  os.Stdin,
+			Stdout: &buf,
+			Stderr: os.Stderr,
+			Tty:    true,
+		})
+		if err == nil {
+			commandOut := buf.String()
+			return commandOut, buf
+		}
+		log.Print(err)
```

Comment:
nit: you should use the `Printf` and the put `\n` after the error output

Reply:
I added Eventually which will take care about possible error. No need to print in